src/node_webstorage.cc

```diff
@@ -0,0 +1,654 @@
+      env(), sqlite3_prepare_v2(_db, sql, -1, &stmt, 0), SQLITE_OK, void());
+  CHECK_ERROR_OR_THROW(env(), sqlite3_step(stmt), SQLITE_DONE, void());
+}
+
+Local<Array> Storage::Enumerate() {
+  if (!Open()) {
+    return Local<Array>();
+  }
+
+  static const char sql[] = "SELECT key FROM nodejs_webstorage";
+  sqlite3_stmt* stmt = nullptr;
+  auto cleanup = OnScopeLeave([&]() { sqlite3_finalize(stmt); });
+  int r = sqlite3_prepare_v2(_db, sql, -1, &stmt, 0);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, Local<Array>());
+  std::vector<Local<Value>> values;
+  while ((r = sqlite3_step(stmt)) == SQLITE_ROW) {
+    CHECK(sqlite3_column_type(stmt, 0) == SQLITE_BLOB);
+    values.emplace_back(
+        String::NewFromUtf8(
+            env()->isolate(),
+            reinterpret_cast<const char*>(sqlite3_column_blob(stmt, 0)),
+            v8::NewStringType::kNormal,
+            static_cast<int>(sqlite3_column_bytes(stmt, 0))).ToLocalChecked());
+  }
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_DONE, Local<Array>());
+  return Array::New(env()->isolate(), values.data(), values.size());
+}
+
+Local<Value> Storage::Length() {
+  if (!Open()) {
+    return Local<Value>();
+  }
+
+  static const char sql[] = "SELECT count(*) FROM nodejs_webstorage";
+  sqlite3_stmt* stmt = nullptr;
+  auto cleanup = OnScopeLeave([&]() { sqlite3_finalize(stmt); });
+  int r = sqlite3_prepare_v2(_db, sql, -1, &stmt, 0);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, Local<Value>());
+  CHECK_ERROR_OR_THROW(env(), sqlite3_step(stmt), SQLITE_ROW, Local<Value>());
+  CHECK(sqlite3_column_type(stmt, 0) == SQLITE_INTEGER);
+  int result = sqlite3_column_int(stmt, 0);
+  return Integer::New(env()->isolate(), result);
+}
+
+Local<Value> Storage::Load(Local<Name> key) {
+  if (key->IsSymbol()) {
+    auto symbol_map = _symbols.Get(env()->isolate());
+    MaybeLocal<Value> result = symbol_map->Get(env()->context(), key);
+    return result.FromMaybe(Local<Value>());
+  }
+
+  if (!Open()) {
+    return Local<Value>();
+  }
+
+  static const char sql[] =
+      "SELECT value FROM nodejs_webstorage WHERE key = ? LIMIT 1";
+  sqlite3_stmt* stmt = nullptr;
+  auto cleanup = OnScopeLeave([&]() { sqlite3_finalize(stmt); });
+  int r = sqlite3_prepare_v2(_db, sql, -1, &stmt, 0);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, Local<Value>());
+  node::Utf8Value utf8key(env()->isolate(), key);
+  r = sqlite3_bind_blob(
+      stmt, 1, utf8key.out(), utf8key.length(), SQLITE_STATIC);
+  CHECK_ERROR_OR_THROW(env(), r, SQLITE_OK, Local<Value>());
+
+  auto value = Local<Value>();
+  r = sqlite3_step(stmt);
+  if (r == SQLITE_ROW) {
+    CHECK(sqlite3_column_type(stmt, 0) == SQLITE_BLOB);
+    value = String::NewFromUtf8(
+        env()->isolate(),
+        reinterpret_cast<const char*>(sqlite3_column_blob(stmt, 0)),
+        v8::NewStringType::kNormal,
+        static_cast<int>(sqlite3_column_bytes(stmt, 0))).ToLocalChecked();
+  } else if (r != SQLITE_DONE) {
+    THROW_SQLITE_ERROR(env(), r);
+  }
```

Comment:
non-blocking nit: There is a fair amount of duplicated boilerplate around the preparation of these statements. Could this be simplified a bit?